fastly/resource_fastly_service_dictionary_items_v1.go

```diff
@@ -0,0 +1,219 @@
+package fastly
+
+import (
+	"fmt"
+	gofastly "github.com/fastly/go-fastly/fastly"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceServiceDictionaryItemsV1() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceServiceDictionaryItemsV1Create,
+		Read:   resourceServiceDictionaryItemsV1Read,
+		Update: resourceServiceDictionaryItemsV1Update,
+		Delete: resourceServiceDictionaryItemsV1Delete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"service_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Service Id",
+			},
+
+			"dictionary_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Dictionary Id",
+			},
+
+			"items": {
+				Type:        schema.TypeMap,
+				Description: "Dictionary Items",
+				Optional:    true,
+			},
+		},
+	}
+}
+
+func resourceServiceDictionaryItemsV1Create(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*FastlyClient).conn
+
+	serviceID := d.Get("service_id").(string)
+	dictionaryID := d.Get("dictionary_id").(string)
+	items := d.Get("items").(map[string]interface{})
+
+	for k, v := range items {
+		_, err := conn.CreateDictionaryItem(&gofastly.CreateDictionaryItemInput{
+			Service:    serviceID,
+			Dictionary: dictionaryID,
+			ItemKey:    k,
+			ItemValue:  v.(string),
+		})
+
+		if errRes, ok := err.(*gofastly.HTTPError); ok {
+			if errRes.StatusCode != 409 {
+				return err
+			}
+		} else if err != nil {
+			return err
+		}
+	}
+
+	// TODO review this
+	d.SetId(fmt.Sprintf("%s/%s", serviceID, dictionaryID))
+	return resourceServiceDictionaryItemsV1Read(d, meta)
+}
+
+func resourceServiceDictionaryItemsV1Update(d *schema.ResourceData, meta interface{}) error {
+
+	conn := meta.(*FastlyClient).conn
+
+	serviceID := d.Get("service_id").(string)
+	dictionaryID := d.Get("dictionary_id").(string)
+
+	d.Partial(true)
+
+	if d.HasChange("items") {
+
+		o, n := d.GetChange("items")
+
+		os := o.(map[string]interface{})
+		ns := n.(map[string]interface{})
+
+		// Handle Removal
+		for k := range os {
+			if _, ok := ns[k]; !ok {
+				err := conn.DeleteDictionaryItem(&gofastly.DeleteDictionaryItemInput{
```

Comment:
As per my above comment, it would be better if this could use the batch endpoint. However it seems that the `go-fastly` client lacks a `DeleteDictionaryItems` method. 😢

Reply:
This function now uses the batch operation provided by the go-fastly client.